src/main/java/oop/animals/enclosures/EnclosureClone.java

```diff
@@ -0,0 +1,26 @@
+package oop.animals.enclosures;
+
+import oop.animals.model.mammals.artiodactyls.Cow;
+
+public class EnclosureClone {
+
+    public Enclosure<Cow> cloneEnclosureOfCows(Enclosure<Cow> enclosureToClone) throws CloneNotSupportedException {
+        Enclosure<Cow> clonedEnclosure = new Enclosure<>();
+        Cow clonedCow;
```

Comment:
You don't need to use variable here, you can use `clonedEnclosure.addAnimal(cow.clone())` instead.